app/app.rb

```diff
@@ -0,0 +1,77 @@
+ENV["RACK_ENV"] ||= "development"
+
+require 'sinatra/base'
+require_relative 'data_mapper_setup'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+
+   use Rack::MethodOverride
+
+  register Sinatra::Flash
+
+  enable :sessions
+  #set :session_secret, 'super secret'
+
+  get '/' do
+    @messages = Message.all.reverse
+    erb :index
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :sign_up
+  end
+
+  post '/users' do
+    @user = User.create(name: params[:name],
```

Comment:
Consider refactoring some of this into a class method on the user model - https://github.com/makersacademy/chitter-challenge/blob/master/docs/review.md#refactor-long-controller-methods